wordpress-mobile/release-toolkit-gutenberg-mobile
# CLI: Fix nvm setup

Fixes #240 

This sets up the npm command to correctly use nvm. 
It also provides a clear space to introduce other node managers.

## Testing 

Follow the [Testing](https://github.com/wordpress-mobile/release-toolkit-gutenberg-mobile/blob/trunk/gbm-cli/Testing.md) guide to set up the tool to use forked repos.

Prerequisite: nvm installed 

### Testing with nvm
- Verify nvm is not loaded in `bash_profile` , e.g.
```
$ bash -l 
$ nvm  
bash: nvm: command not found
```
- Make sure your default node in nvm is not v20
- Try running `go run main.go release prepare gb {Version}` 
- Verify that the command reaches the "create PR" confirmation step

### Testing without nvm
- Make sure your default node in nvm is not v20
- Try `NVM_DIR= go run main.go release prepare gb {Version}`
- This should fail with unsupported engine error
- Set the default node to v20 e.g. `nvm alias default v20`
- Try `NVM_DIR= go run main.go release prepare gb {Version}` again 
- Verify that the command reaches the "create PR" confirmation step
